webrtc/src/ice_transport/ice_gatherer.rs

```diff
@@ -84,8 +85,7 @@ impl RTCIceGatherer {
         // ensure we do not do anything expensive other than the actual agent
         // creation in this function.
 
-        let mut agent = self.agent.lock().await;
```

Comment:
This is a regression, we create duplicated agents again. See the comment above this

Reply:
Fixed